internal/vault/config.go

```diff
@@ -96,6 +98,9 @@ func MakeVaultClient(ctx context.Context, cfg *ClientConfig, client ctrlclient.C
 	if cfg.VaultNamespace != "" {
 		c.SetNamespace(cfg.VaultNamespace)
 	}
+	for k, v := range cfg.Headers {
```

Comment:
I think we should move the custom header setting to come before line 98. That will ensure that `spec.headers` does not override `cfg.VaultNamespace` by setting `X-Vault-Namespace`.

Reply:
Sure, also made setting the namespace in VaultConnection an error condition in [10af3b4](https://github.com/hashicorp/vault-secrets-operator/pull/629/commits/10af3b436e5eecb59f858ab6fb1156e3493e936e)